ethcore/src/trace/bloom.rs

```diff
@@ -0,0 +1,119 @@
+use bloomchain::Bloom;
+use bloomchain::group::{BloomGroup, GroupPosition};
+use util::rlp::*;
+use util::{H256};
+use basic_types::LogBloom;
+
+/// Helper structure representing bloom of the trace.
+#[derive(Clone)]
+pub struct BlockTracesBloom(LogBloom);
+
+impl From<LogBloom> for BlockTracesBloom {
+	fn from(bloom: LogBloom) -> BlockTracesBloom {
+		BlockTracesBloom(bloom)
+	}
+}
+
+impl From<Bloom> for BlockTracesBloom {
+	fn from(bloom: Bloom) -> BlockTracesBloom {
+		let bytes: [u8; 256] = bloom.into();
+		BlockTracesBloom(LogBloom::from(bytes))
+	}
+}
+
+impl Into<Bloom> for BlockTracesBloom {
+	fn into(self) -> Bloom {
+		let log = self.0;
+		Bloom::from(log.0)
+	}
+}
+
+/// Represents group of X consecutive blooms.
+#[derive(Clone)]
+pub struct BlockTracesBloomGroup {
+	blooms: Vec<BlockTracesBloom>,
+}
+
+impl From<BloomGroup> for BlockTracesBloomGroup {
+	fn from(group: BloomGroup) -> Self {
+		let blooms = group.blooms
+			.into_iter()
+			.map(From::from)
+			.collect();
+
+		BlockTracesBloomGroup {
+			blooms: blooms
+		}
+	}
+}
+
+impl Into<BloomGroup> for BlockTracesBloomGroup {
+	fn into(self) -> BloomGroup {
+		let blooms = self.blooms
+			.into_iter()
+			.map(Into::into)
+			.collect();
+
+		BloomGroup {
+			blooms: blooms
+		}
+	}
+}
+
+impl Decodable for BlockTracesBloom {
+	fn decode<D>(decoder: &D) -> Result<Self, DecoderError> where D: Decoder {
+		Decodable::decode(decoder).map(BlockTracesBloom)
+	}
+}
+
+impl Encodable for BlockTracesBloom {
+	fn rlp_append(&self, s: &mut RlpStream) {
+		Encodable::rlp_append(&self.0, s)
+	}
+}
+
+impl Decodable for BlockTracesBloomGroup {
+	fn decode<D>(decoder: &D) -> Result<Self, DecoderError> where D: Decoder {
+		let blooms = try!(Decodable::decode(decoder));
+		let group = BlockTracesBloomGroup {
+			blooms: blooms
+		};
+		Ok(group)
+	}
+}
+
+impl Encodable for BlockTracesBloomGroup {
+	fn rlp_append(&self, s: &mut RlpStream) {
+		Encodable::rlp_append(&self.blooms, s)
+	}
+}
+
+/// Represents BloomGroup position in database.
+#[derive(PartialEq, Eq, Hash, Clone)]
+pub struct TraceGroupPosition {
+	/// Bloom level.
+	pub level: usize,
+	/// Group index.
+	pub index: usize,
+}
+
+impl From<GroupPosition> for TraceGroupPosition {
+	fn from(p: GroupPosition) -> Self {
+		TraceGroupPosition {
+			level: p.level,
+			index: p.index,
+		}
+	}
+}
+
+impl TraceGroupPosition {
+	/// Returns trace group position represented as a 256 bit hash.
+	pub fn hash(&self) -> H256 {
+		use std::ptr;
+		let mut hash = H256::default();
+		unsafe {
+			ptr::copy(&[self.level, self.index] as *const usize as *const u8, hash.as_mut_ptr(), 16);
```

Comment:
This won't work for 32-bit build. usize is pointer sized. 
Why not just use two 32-bit values and combine them into one 64-bit and return it?